sycl/doc/extensions/proposed/sycl_ext_oneapi_prefetch.asciidoc

```diff
@@ -0,0 +1,454 @@
+levels. Any such alterations are expected to be documented in the definition
+of those new hints.
+====
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+enum class cache_level {
+  L1,
+  L2,
+  L3,
+  L4,
+};
+
+struct nontemporal;
+
+struct prefetch_hint_key {
+  template <cache_level Level, typename Hint>
+  using value_t = property_value<prefetch_hint_key,
+                                 std::integral_constant<cache_level, Level>,
+                                 Hint>
+};
+
+template <cache_level Level, typename Hint>
+inline constexpr prefetch_hint_key::value_t<Level, Hint> prefetch_hint;
+
+inline constexpr prefetch_hint_key::value_t<cache_level::L1, void> prefetch_hint_L1;
+inline constexpr prefetch_hint_key::value_t<cache_level::L2, void> prefetch_hint_L2;
+inline constexpr prefetch_hint_key::value_t<cache_level::L3, void> prefetch_hint_L3;
+inline constexpr prefetch_hint_key::value_t<cache_level::L4, void> prefetch_hint_L4;
+
+inline constexpr prefetch_hint_key::value_t<cache_level::L1, nontemporal> prefetch_hint_L1_NT;
+inline constexpr prefetch_hint_key::value_t<cache_level::L2, nontemporal> prefetch_hint_L2_NT;
+inline constexpr prefetch_hint_key::value_t<cache_level::L3, nontemporal> prefetch_hint_L3_NT;
+inline constexpr prefetch_hint_key::value_t<cache_level::L4, nontemporal> prefetch_hint_L4_NT;
+
+} // namespace sycl::ext::oneapi::experimental
+----
+
+If the `nontemporal` type is used as the value of the `Hint` template
+parameter, this acts as a hint that the data being prefetched will not be
+reused.
+
+[NOTE]
+====
+The implementation of `nontemporal` is device-specific, owing to differences in
+cache hierarchies and replacement policies. On some devices non-temporal data
+may be evicted earlier than temporal data, while on other devices non-temporal
+data may not be stored in certain levels of cache at all.
+====
+
+
+=== Work-item prefetches
+
+The functions in this section allow individual work-items to prefetch data.
+
+[source,c++]
+----
+namespace sycl::ext::oneapi::experimental {
+
+template <typename Properties = ext::oneapi::experimental::properties<>>
+void prefetch(void* ptr, Properties properties = {});
+
+template <typename Properties = ext::oneapi::experimental::properties<>>
+void prefetch(void* ptr, size_t bytes, Properties properties = {});
+
+template <typename T, typename Properties = ext::oneapi::experimental::properties<>>
+void prefetch(T* ptr, Properties properties = {});
+
+template <typename T, typename Properties = ext::oneapi::experimental::properties<>>
+void prefetch(T* ptr, size_t count, Properties properties = {});
+
+// Only available if Accessor::dimensions == 1 and Accessor::access_mode != write
+template <typename Accessor, typename Properties = ext::oneapi::experimental::properties<>>
+void prefetch(Accessor* acc, size_t offset, Properties properties = {});
+
+// Only available if Accessor::dimensions == 1 and Accessor::access_mode != write
```

Comment:
Rather than saying `Accessor::access_mode != write`, I think we should say that access mode can be either `read` or `read_write`.  This excludes all of the deprecated modes like `atomic` and `discard_write`.

Reply:
Fixed in e52b5725a59a.